src/spring2024/my_ptr.cpp

```diff
@@ -0,0 +1,242 @@
+#include <iostream>
+#include <memory>
+#include <utility>
+
+// This file contains codes used in the Spring2024 15445/645 C++ bootcamp.
+// It dives deeply into C++ new features like move constructor/assignment, move semantics, unique_ptr, shared_ptr,
+// wrapper class, etc., by implementing a simple version of unique_ptr from scratch.
+// **IMPORTANT**: please BEGIN reading from MAIN function!
+
+// Notice: it is just a simple version of std::unique_pointer<T>. The real implementation is much more complex!
+// A template allows us to replace any type T, with what we want later in our code.
+template <typename T>
+class Pointer {
+ public:
+  Pointer(T val) {
+    ptr_ = new T;
```

Comment:
does not work for types w/o default constructor, but it's fine for educational purposes...

Reply:
Updated. Thx!